lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -1916,6 +1939,30 @@ void OCLToSPIRVBase::visitCallLdexp(CallInst *CI, StringRef MangledName,
   visitCallBuiltinSimple(CI, MangledName, DemangledName);
 }
 
+void OCLToSPIRVBase::visitConvertBFloat16AsUshort(CallInst *CI,
+                                                  StringRef DemangledName) {
+  assert(CI->getCalledFunction() && "Unexpected indirect call");
```

Comment:
Maybe no need for this assert as this condition is checked in `visitCallInst`:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/56cad67cb57e5b90011b3e8cecd3636b954d59d6/lib/SPIRV/OCLToSPIRV.cpp#L395-L399

Reply:
I agree, there is no need for this assert. About the negative test - I've checked, the translator fails with:
> virtual void SPIRV::SPIRVTypeVector::validate() const: Assertion `CompCount == 2 || CompCount == 3 || CompCount == 4 || CompCount == 8 || CompCount == 16' failed

According to Mikhail [comment](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1406#issuecomment-1041392080) I think it will be enough to check that the function name matches the returning type and the argument type and add the appropriate test. I will add it a littlie bit later, thanks.

Reply:
Done in [9139158](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1406/commits/9139158c5a3936529c38fb66d1157eabd9a3c62b) ... [c368af5](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1406/commits/c368af564860f325bd09b5c8a1b9004cc6f303b2).